docs: Update Wave Build (#995)
seqeralabs/wave: docs/install/configure-wave-build.md

title: Configure Wave build
---

This guide covers extending your existing Wave installation on Kubernetes to support container build capabilities. This enables Wave's full feature set, including container building, freezing, security scanning, mirroring, and advanced caching.

> Wave builds use a **two-namespace architecture**: the main Wave application runs in one namespace (e.g., `wave`) and spawns ephemeral build, scan, and mirror pods in a dedicated build namespace (e.g., `wave-build`). Both namespaces share a single EFS filesystem so that build artifacts are accessible to both the app and the build pods.

Reviewer: ```suggestion
Wave builds use a **two-namespace architecture**. The main Wave application runs in one namespace (e.g., `wave`) and spawns ephemeral build, scan, and mirror pods in a dedicated build namespace (e.g., `wave-build`). Both namespaces share a single EFS filesystem, allowing build artifacts to be accessible to both the application and the build pods.
```

                       │                                  │
                       └──────────┬───────────────────────┘
                                  ▼
                       ┌─────────────────────┐
                       │  AWS EFS filesystem  │
                       │  (shared storage)    │
                       └─────────────────────┘
```

## Prerequisites

Before extending Wave for build support, ensure you have:

- **Existing Wave installation** — Basic Wave deployment already running in augmentation-only mode (see [Kubernetes installation](./kubernetes.md))
- **AWS EKS cluster** — Build capabilities require AWS-specific integrations (EFS, IRSA, ECR)
- **[AWS EFS CSI Driver](https://docs.aws.amazon.com/eks/latest/userguide/efs-csi.html)** — Installed on the cluster. Verify with: `kubectl get pods -n kube-system | grep efs`
- **AWS EFS filesystem** — Provisioned in the same VPC as your EKS cluster (see [Create the EFS filesystem](#create-the-efs-filesystem))
- **IAM role for IRSA** — With permissions for ECR, S3, and SES (see [Configure IAM role](#configure-iam-role))
- **Cluster admin permissions** — Required to create namespaces, RBAC policies, and storage resources

> - **CNI with NetworkPolicy support** (optional) — Such as Calico or Cilium, required if you want to enforce [build pod network policies](../configure-wave.md#build-pod-network-policies)

Reviewer: ```suggestion
:::info[**Prerequisites**]
You will need the following to get started:

- An existing Wave deployment running in augmentation-only mode
    - See [Kubernetes installation](./kubernetes.md) for more information
- An AWS EKS cluster
- An [AWS EFS CSI Driver](https://docs.aws.amazon.com/eks/latest/userguide/efs-csi.html)
- An AWS EFS filesystem provisioned in the same VPC as your EKS cluster
    - See [Create the EFS filesystem](#create-the-efs-filesystem) for more information
- An IAM role for IRSA, with permissions for ECR, S3, and SES
    - See [Configure IAM role](#configure-iam-role) for more information
- Cluster admin permissions to create RBAC policies and storage resources
- CNI with NetworkPolicy support (optional; e.g., Calico or Cilium are required to enforce [build pod network policies](../configure-wave.md#build-pod-network-policies))
:::
```
We are slowly replacing the titles with admonitions like this.
